src/UiExplorer.elm

```diff
@@ -221,10 +225,50 @@ nextPage id config (PageBuilder previous) =
         , update = update_
         , view = view_
         , subscriptions = subscriptions_
-        , ids = id :: previous.ids
+        , ids = { pageId = id, pageGroup = previous.pageGroup } :: previous.ids
+        , pageGroup = previous.pageGroup
         }
 
 
+{-| If your list of pages on the sidebar is starting to get too long, you can group some of them together with `groupPages`.
+
+    import MyPages
+    import UiExplorer
+
+    flatPages =
+        UiExplorer.firstPage "Login" MyPage.login
+            |> UiExplorer.nextPage "About us" MyPage.aboutUs
+            |> UiExplorer.nextPage "Fonts" MyPage.fonts
+            |> UiExplorer.nextPage "Colors" MyPage.colors
+            |> UiExplorer.nextPage "Basic components" MyPage.basicComponents
+            |> UiExplorer.nextPage "Homepage" MyPage.homepage
+
+    nowItsGrouped =
+        UiExplorer.firstPage "Login" MyPage.login
+            |> UiExplorer.nextPage "About us" MyPage.aboutUs
+            |> UiExplorer.groupPages "Building blocks"
+                (UiExplorer.nextPage "Fonts" MyPage.fonts
+                    >> UiExplorer.nextPage "Colors" MyPage.colors
+                    >> UiExplorer.nextPage "Basic components" MyPage.basicComponents
+                )
+            |> UiExplorer.nextPage "Homepage" MyPage.homepage
+
+Two things to note:
+
+  - Normally pages need unique names, but with groups it's okay to have two pages use the same name so long as they are in different groups.
+  - Due to [`firstPage`](#firstPage) having a different type signature from [`nextPage`](#nextPage), you can't place the first page in a group.
+    If this is a problem, [create an issue on github](https://github.com/insurello/elm-ui-explorer/issues/new) explaining your use case.
```

Comment:
To me this is an annoying limitation. But I'm hoping in practice no one will care.

One other problem not mentioned in the documentation is that currently, nested groups don't look so good in the UI. I couldn't come up with a good solution for indicating how groups should be nested in the sidebar. Currently in our example app it looks like this: 
<img width="278" alt="image" src="https://user-images.githubusercontent.com/5068391/80178223-fdd47200-85fd-11ea-9430-72fefc4bea47.png">
`Signup and login` is a top level group and `Login` is a nested group. You might not realize that at a glance though given there's no indentation. I don't want to add indentation though because pretty quickly the page buttons won't fit on the sidebar without horizontal scrolling. I also tried having those vertical lines you see in code editors that indicate hierarchy in code but I couldn't get that to look good. For now hopefully there one be much need for nested groups?

Reply:
Maybe you can put a semi-transparent background on expanded groups (on the parent and all its children), making each expanded group darker and darker (or lighter and lighter)?

Reply:
Good point. That seems to work quite well.
<img width="222" alt="image" src="https://user-images.githubusercontent.com/5068391/80186083-bb1a9600-860d-11ea-9db0-3e2f79eda5b5.png">

Reply:
- Should we add the background only when expanded? (See Footer in your screenshot.) Not sure.
- Why does `Login (width 1000)` have a different background color, but not triangle?

Reply:
* To me at least it seems more clear how things are grouped if the group button always has matches the color of its contents. It also makes it more clear that it's a group button and not a page button.
* `Login (width 1000)` is the page I've selected (selected page buttons are blue).

Reply:
I went ahead and made the group color a little less intense 
<img width="227" alt="image" src="https://user-images.githubusercontent.com/5068391/80190006-e1dbcb00-8613-11ea-9070-026856796b8d.png">

Reply:
Ah, makes sense. :+1: